Test failure (`test_zzphase`)

Running this test on current `develop`:

```
    @pytest.mark.skipif(skip_remote_tests, reason=REASON)
    @pytest.mark.parametrize(
        "authenticated_quum_backend", [{"device_name": "H1-1SC"}], indirect=True
    )
    def test_zzphase(
        authenticated_quum_backend: QuantinuumBackend,
    ) -> None:
        backend = authenticated_quum_backend
        c = Circuit(2, 2, "test rzz")
        c.H(0)
        c.CX(0, 1)
        c.Rz(0.3, 0)
        c.CY(0, 1)
        c.ZZPhase(0.1, 1, 0)
        c.measure_all()
        c0 = backend.get_compiled_circuit(c, 0)
    
>       assert c0.n_gates_of_type(OpType.ZZPhase) > 0
E       assert 0 > 0
E        +  where 0 = <bound method PyCapsule.n_gates_of_type of [Rz(1) q[0]; Rz(1) q[1]; PhasedX(0.5, 0.5) q[0]; PhasedX(0.5, 1) q[1]; Rz(3...0) q[1]; Rz(1.5) q[0]; PhasedX(0.5, 1.5) q[1]; Measure q[1] --> c[1]; PhasedX(0.5, 1.5) q[0]; Measure q[0] --> c[0]; ]>(<OpType.ZZPhase: 68>)
E        +    where <bound method PyCapsule.n_gates_of_type of [Rz(1) q[0]; Rz(1) q[1]; PhasedX(0.5, 0.5) q[0]; PhasedX(0.5, 1) q[1]; Rz(3...0) q[1]; Rz(1.5) q[0]; PhasedX(0.5, 1.5) q[1]; Measure q[1] --> c[1]; PhasedX(0.5, 1.5) q[0]; Measure q[0] --> c[0]; ]> = [Rz(1) q[0]; Rz(1) q[1]; PhasedX(0.5, 0.5) q[0]; PhasedX(0.5, 1) q[1]; Rz(3.5) q[0]; PhasedX(0.5, 0) q[1]; PhasedX(0.5... 0) q[1]; Rz(1.5) q[0]; PhasedX(0.5, 1.5) q[1]; Measure q[1] --> c[1]; PhasedX(0.5, 1.5) q[0]; Measure q[0] --> c[0]; ].n_gates_of_type
E        +    and   <OpType.ZZPhase: 68> = OpType.ZZPhase

backend_test.py:547: AssertionError
```

--- a/tests/backend_test.py
+++ b/tests/backend_test.py
@@ -544,7 +544,10 @@ def test_zzphase(
     c.measure_all()
     c0 = backend.get_compiled_circuit(c, 0)
 
-    assert c0.n_gates_of_type(OpType.ZZPhase) > 0
+    if OpType.ZZPhase in backend._gate_set:
+        assert c0.n_gates_of_type(OpType.ZZPhase) > 0
+    else:
+        assert c0.n_gates_of_type(OpType.ZZMax) > 0
 
     n_shots = 4
     handle = backend.process_circuits([c0], n_shots)[0]

